### ISSUE TITLE
Appimage won't work anymore

### ISSUE THREAD

**Issue by illusioon** [PaulCombal/SamRewritten-legacy#100]
./SamRewritten-x86_64.AppImage /tmp/.mount_SamRewrxECfA/usr/bin/samrewritten: symbol lookup error: /usr/lib/libgio-2.0.so.0: undefined symbol: g_module_open_full

**Comment by PaulCombal** [comment]
Did it work before? Did it break after an update?

What's your OS and version?

**Comment by illusioon** [comment]
Yes, it used to work, but after an update it stopped maybe because glib or glibc got updated. 

My distro is void linux x86_64, kernel 5.13.19_1 , glib-2.70.0_1, glibc-2.32_2

**Comment by PaulCombal** [comment]
Can you build it on your machine? If you produce an AppImage yourself, does it work?

**Comment by illusioon** [comment]
Sorry to take so long to answer I was busy with my job.

So I build it on my machine and it works. Compiled from source(following the instructions README.MD) and then build a app image with "build_appimage.sh"

**Comment by nolddor** [comment]
Same problem here using:

- Debian GNU/Linux bookworm/sid x86_64 
- Kernel: 5.14.0-4-amd64 
- glib2.0 (2.70.1-1) 

SamRewritten worked last time i tried but not anymore...

**Comment by PaulCombal** [comment]
Does it work when you compile it yourself? If so feel free to upload it here to share with others

**Comment by nolddor** [comment]
Yes, building mine works like a charm. Could you somehow, host my build within your repository as an artifact?
https://we.tl/t-GE5qmDwK2E (will be deleted after 7 days)

**Comment by PaulCombal** [comment]
Well your version doesn't work on my machine.. your GLIBC version is newer than mine so it crashes on startup. To be fair I don't have the energy to look into this issue, but would a Snap/flatpak version of SamRewritten would be a good alternative?

Anyway here's a long lasting link for your build

[SamRewritten-Debian_bookworm-2a9fe6f-x86_64.AppImage.zip](https://github.com/PaulCombal/SamRewritten/files/7590953/SamRewritten-Debian_bookworm-2a9fe6f-x86_64.AppImage.zip)

**Comment by nolddor** [comment]
A snap version of SamRewritten would indeed be really helpful!

**Comment by orowith2os** [comment]
How goes on the alternative packaging of this? Is there any progress on getting it to work with the Steam flatpak, such as running the binaries in the same flatpak environment?

AFAIK SteamTinkerLaunch is packaged as a Steam flatpak runtime extension to allow it to interact with the Steam flatpak, so I'm thinking SamRewritten would need to do the same to work with it.

**Comment by PaulCombal** [comment]
@orowith2os as I said [here](https://github.com/PaulCombal/SamRewritten/issues/135#issuecomment-1328057838), I'm waiting for a package manager to handle all the main dependencies to switch to it, then review the whole CI/CD pipeline

**Comment by qurious-pixel** [comment]
excluding `libgmodule-2.0.so.0` is enough for it to run. 

```
LD_LIBRARY_PATH=AppDir/usr/lib ./linuxdeploy-x86_64.AppImage --appdir AppDir --exclude-library=libgmodule-2.0 --output appimage
```

**Comment by dotbanana** [comment]
Does not work. 
Waste of time really.
